pkg/util/ecs/metadata/v3or4/client_test.go

```diff
@@ -0,0 +1,131 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+//go:build docker
+
+package v3or4
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/testutil"
+)
+
+func TestGetV4TaskWithTags(t *testing.T) {
+	testDataPath := "./testdata/task_with_tags.json"
+	dummyECS, err := testutil.NewDummyECS(
+		testutil.FileHandlerOption("/v4/1234-1/taskWithTags", testDataPath),
+	)
+	require.NoError(t, err)
+	ts := dummyECS.Start()
+	defer ts.Close()
+
+	client := NewClient(fmt.Sprintf("%s/v4/1234-1", ts.URL), "v4")
+	task, err := client.GetTaskWithTags(context.Background())
+	require.NoError(t, err)
+
+	require.Equal(t, *expected(), *task)
+}
+
+// expected generates an expected Task from ./testdata/task.json
+func expected() *Task {
+	return &Task{
+		ClusterName:   "arn:aws:ecs:us-east-1:123457279990:cluster/ecs-cluster",
```

Comment:
`expected` may not need to be a function.

```suggestion
	require.Equal(t, expected, task)
}

// expected generates an expected Task from ./testdata/task.json
var expected := &Task{
		ClusterName:   "arn:aws:ecs:us-east-1:123457279990:cluster/ecs-cluster",
```